src/components/tree-views-container.js

```diff
@@ -352,11 +352,22 @@ const TreeViewsContainer = () => {
                     e.directory === selectedDirectoryRef.current.elementUuid &&
                     current[e.elementName] === undefined
             );
-
             if (elementsToMerge != null && elementsToMerge.length > 0) {
-                return [...current, ...elementsToMerge].sort(function (a, b) {
-                    return a.elementName.localeCompare(b.elementName);
-                });
+                const filtredCurrentElements = current.filter(
```

Comment:
Add a comment
// We need to filter current array of elements in elementsToMerge to avoid duplicates in the directoryContent component.
// An uploading element doesn't have an elementUuid yet, then we filter on element Name and type.